app/models/node.rb

```diff
@@ -48,6 +50,29 @@ def submission_exists_for?(user)
     self.submissions.open.for(user).present?
   end
 
+  # Override in subclasses to add options
+  # Format is {option: :default_value}
+  def available_options
+    {}
+  end
+
+  # 1) reform requires options to be an object with attributes
+  # 2) activerecord wants it to be a hash
+  # 3) Storext's default type coercion doesn't work properly with OpenStruct
+  # so for now, we just do the conversion by hand
+  def options
+    hash = super
+    if hash
+      OpenStruct.new(hash)
```

Comment:
Yeah, possibly. But building the proper abstractions when you only have one is hard. I'd rather wait until there's at least two.